Criterion/Main/Options.hs

```diff
@@ -154,12 +159,14 @@ match :: ReadM MatchType
 match = do
   m <- readerAsk
   case map toLower m of
-    mm | mm `isPrefixOf` "pfx"    -> return Prefix
-       | mm `isPrefixOf` "prefix" -> return Prefix
-       | mm `isPrefixOf` "glob"   -> return Glob
-       | otherwise                ->
-         readerError $ show m ++ " is not a known match type. "
-                              ++ "Try \"prefix\" or \"glob\"."
+    mm | mm `isPrefixOf` "pfx"      -> return Prefix
+       | mm `isPrefixOf` "prefix"   -> return Prefix
+       | mm `isPrefixOf` "glob"     -> return Glob
+       | mm `isPrefixOf` "pattern"  -> return Pattern
```

Comment:
One unfortunate thing about these sequential `isPrefixOf` checks is that `--match p` will pick `prefix` over `pattern`, even though `p` is a prefix of both. Is this acceptable? Or should we make `--match p` give an error which notes the ambiguity?